File: Introductory Problems/Permutations/Permutations.py
def solve(n):
    if n == 2 or n == 3:
        print("NO SOLUTION")
        return
    evens = [i for i in range(1, n + 1, 2)]
    odds = [i for i in range(2, n + 1, 2)]
    series = odds + evens
    print(*series)

if __name__ == "__main__":
    n = int(input())
    solve(n)

File: Introductory Problems/Permutations/Permutations.cpp
#include <iostream>

using namespace std;


void solve(const int& n)
{
    if (n == 2 || n == 3) 
    {
        cout << "NO SOLUTION" << endl;
        return;
    }
    for (int i = 2; i <= n; i+=2)
    {
        cout << i << " ";
    }
    for (int i = 1; i <= n; i+=2)
    {
        cout << i << " ";
    }
}

int main()
{
    int n;
    cin >> n;
    solve(n);
    return 0;
}

